src/components/sidebar/right-sidebar/datepicker/Datepicker.tsx

```diff
@@ -16,15 +21,32 @@ interface DatepickerProps {
 export default function Datepicker({ page }: DatepickerProps) {
   const [state, dispatch] = useContext(AppContext);
   const pageState = state[page as keyof State] as PageState;
-  const [filterStartDate, filterEndDate] = pageState.filters.publish_date;
-  const [startDate, setStartDate] = useState(filterStartDate);
-  const [endDate, setEndDate] = useState(filterEndDate);
+  const [filterStartDate, filterEndDate]: Date[] = Array.from(pageState.filters.publish_date);
+  const [startDate, setStartDate] = useState<Date>(filterStartDate);
+  const [endDate, setEndDate] = useState<Date>(filterEndDate);
+  const minDate = state.calendarStartDates.minDate;
+  const maxDate = state.calendarStartDates.maxDate;
+  const [sliderMax, setSliderMax] = useState<number>((maxDate.getTime()-minDate.getTime())/86400000);
+  const [values, setValues] = useState<Date[]>([startDate, endDate]);

```

Comment:
Can we be a bit clearer with these state variable names to differentiate state variables that pertain to the datepicker vs the slider